src/charm.py

```diff
@@ -99,8 +99,8 @@ def _on_start(self, event: EventBase) -> None:
             servers, unit_config = self.cluster.ready_to_start(self.unit)
         except (NotUnitTurnError, UnitNotFoundError, NoPasswordError) as e:
             logger.info(str(e))
-            # defaults to MaintenanceStatus
             self.unit.status = self.cluster.status
+            time.sleep(2)  # accounts for when defers are used up before leader updates config
```

Comment:
might want to use tenacity instead. You're `event.defer()` might allow another hook to run before re-running `_on_start ` due to how juju handles `defer`.

Reply:
Currently I don't have any listened events for non-leader units, so no other event will trigger.
That said, I'll keep in mind in case we add more events!